backend/src/routes/auth.ts

```diff
@@ -16,9 +17,16 @@ const authRouteSetup = (
 			try {
 				const reqSchema = z.object({
 					body: z.object({
-						username: z.string(),
-						password: z.string(),
-						email: z.string(),
+						username: z.string().max(100),
+						password: z
+							.string()
+							.min(8)
+							.max(72)
+							.refine(
+								(val) => val.match(passwordRegex),
+								"password should have at least 8 characters, and only contain alphanumeric or the following special characters: ~`! @#$%^&*()_-+={[}]|:;\"'<,>.?/",
+							),
+						email: z.string().email().max(100),
```

Comment:
apparently the upper limit for email is 254 characters, but since local-part <= 64 and the most popular domains aren't that long, this should be sufficient

Reply:
we can up the limit if anyone ever complaints.